contracts/service-registry/src/state.rs

```diff
@@ -126,7 +127,7 @@ pub enum AuthorizationState {
 // maps service_name -> Service
 pub const SERVICES: Map<&str, Service> = Map::new("services");
 // maps (service_name, chain_name, worker_address) -> ()
-pub const WORKERS_PER_CHAIN: Map<(&str, &str, &Addr), ()> = Map::new("workers_per_chain");
+pub const WORKERS_PER_CHAIN: Map<(&ChainName, &str, &Addr), ()> = Map::new("workers_per_chain");
```

Comment:
I'd be curious if this change actually increases gas cost for `DeclareChainSupport`, because it changes the structure of the map. Before, every insert in that method shared the same first element of the primary key (service name), but now the first element is different for every insert. So it will be worse cache performance.

I'd prefer to keep the order the same as before. If you have to define some additional things for the compiler to be happy, that's ok.